receiver/awsfirehosereceiver/logs_receiver.go

```diff
@@ -61,8 +55,32 @@ func newLogsReceiver(
 	}, nil
 }
 
-// Consume uses the configured unmarshaler to unmarshal each record
-// into a plog.Logs and pass it to the next consumer, one record at a time.
+// Start sets the consumer's log unmarshaler to either a built-in
+// unmarshaler or one loaded from an encoding extension.
+func (c *logsConsumer) Start(_ context.Context, host component.Host) error {
+	encoding := c.config.Encoding
+	if encoding == "" {
+		encoding = c.config.RecordType
+		if encoding == "" {
+			encoding = defaultLogsEncoding
+		}
+	}
+	if encoding == cwlog.TypeStr {
+		// TODO: make cwlogs an encoding extension
+		c.unmarshaler = cwlog.NewUnmarshaler(c.settings.Logger)
+	} else {
+		unmarshaler, err := loadEncodingExtension[plog.Unmarshaler](host, encoding, "logs")
+		if err != nil {
+			return err
```

Comment:
Can we augment errors with more context, here and elsewhere?

```suggestion
			return fmt.Errorf("failed to load encoding extension: %w", err)
```